api/src/certification/session/domain/usecases/get-session-certification-candidates.js

```diff
@@ -1,13 +1,55 @@
 /**
- * @typedef {import('../../../shared/domain/usecases/index.js').CertificationCandidateRepository} CertificationCandidateRepository
+ * @typedef {import('../../../shared/domain/usecases/index.js').CandidateRepository} CandidateRepository
+ * @typedef {import('../../../shared/domain/usecases/index.js').ComplementaryCertificationRepository} ComplementaryCertificationRepository
+ * @typedef {import('../../../../shared/domain/errors.js').NotFoundError} NotFoundError
  */
+import { EnrolledCandidate } from '../read-models/EnrolledCandidate.js';
+import bluebird from 'bluebird';
 
 /**
+ * @function
  * @param {Object} params
- * @param {CertificationCandidateRepository} params.certificationCandidateRepository
+ * @param {CandidateRepository} params.candidateRepository
+ * @param {ComplementaryCertificationRepository} params.complementaryCertificationRepository
+ * @returns {EnrolledCandidate}
+ * @throws {NotFoundError} a candidate is linked to an unexisting complementary certification
  */
-const getSessionCertificationCandidates = async function ({ sessionId, certificationCandidateRepository }) {
-  return certificationCandidateRepository.findBySessionId(sessionId);
+const getSessionCertificationCandidates = async function ({
+  sessionId,
+  candidateRepository,
+  complementaryCertificationRepository,
+}) {
+  const candidates = await candidateRepository.findBySessionId(sessionId);
+
+  const complementaryCertifications = await getUniqComplementaryCertifications({
+    candidates,
+    complementaryCertificationRepository,
+  });
+
+  return candidates.map((candidate) => {
+    return EnrolledCandidate.fromCandidateAndComplementaryCertification({
+      candidate,
+      complementaryCertification: complementaryCertifications.find(
+        (complementaryCertification) => complementaryCertification?.id === candidate.complementaryCertificationId,
+      ),
+    });
+  });
 };
 
 export { getSessionCertificationCandidates };
+
+const ignoreIfNoComplementaryCertification = (candidate) => !!candidate.complementaryCertificationId;
+const getUniqComplementaryCertificationIds = (uniq, candidate) => uniq.add(candidate.complementaryCertificationId);
```

Comment:
Le nom correspond au besoin mais pas à l'implem:
```suggestion
const addComplementaryCertificationIdFromCandidate = (set, candidate) => set.add(candidate.complementaryCertificationId);
```

Reply:
Je me rends compte que j'ai très souvent ce soucis là 😅

Reply:
Done